dns_cache/storage.go

```diff
@@ -0,0 +1,81 @@
+package dns_cache
+
+import (
+	"net"
+	"reflect"
+	"time"
+
+	cache "github.com/pmylund/go-cache"
+)
+
+type DnsCacheItem struct {
+	addrs []string
+}
+
+func (item *DnsCacheItem) IsEqualsTo(addrs []string) bool {
```

Comment:
generally we don't add to main code things like this - code pieces needed to run unit-tests